xonsh/prompt/vc.py

```diff
@@ -52,23 +52,33 @@ def get_git_branch():
         branch = None
     return branch
 
+def _get_hg_root(q):
+    _curpwd = os.getcwd()
+    while True:
+        if any([b.name == '.hg' for b in os.scandir(_curpwd)]):
+            q.put(_curpwd)
+            break
+        else:
+            _oldpwd = _curpwd
+            _curpwd = os.path.split(_curpwd)[0]
+            if _oldpwd == _curpwd:
+                return False
```

Comment:
What's strange to me is that this is how I originally implemented this:

https://github.com/xonsh/xonsh/pull/239/files#diff-b08c376dd29024ccd6f9aef90f2ea96fR220

But for some reason it was subsequently changed. I have vague memories of someone complaining about issues on Windows, but I can't find the pull request where this behavior was subsequently changed.